test/controllers/works_controller_test.rb

```diff
@@ -4,16 +4,31 @@
   describe "root" do
     it "succeeds with all media types" do
       # Precondition: there is at least one media of each category
+      %w(albums books movies).each do |category|
+        Work.by_category(category).length.must_be :>,0
 
-    end
+        get root_path
 
+      end
+    end
     it "succeeds with one media type absent" do
       # Precondition: there is at least one media in two of the categories
+      %w(albums movies).each do |category|
+        Work.by_category(category).length.must_be :>,0
+        Work.by_category(:books).destroy_all
```

Comment:
This is strange to  me - inside the loop you're both checking that the category has members and destroying all the books. Probably the `destroy_all` wants to be outside the loop.